x/leverage/keeper/keeper.go

```diff
@@ -6,26 +6,31 @@ import (
 	"github.com/cosmos/cosmos-sdk/codec"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	simcos "github.com/cosmos/cosmos-sdk/x/simulation"
 	"github.com/tendermint/tendermint/libs/log"
-
+	"github.com/umee-network/umee/x/leverage/simulation"
 	"github.com/umee-network/umee/x/leverage/types"
 )
 
 type Keeper struct {
-	cdc          codec.Codec
-	storeKey     sdk.StoreKey
-	paramSpace   paramtypes.Subspace
-	hooks        types.Hooks
-	bankKeeper   types.BankKeeper
-	oracleKeeper types.OracleKeeper
+	cdc           codec.Codec
+	storeKey      sdk.StoreKey
+	paramSpace    paramtypes.Subspace
+	hooks         types.Hooks
+	accountKeeper simcos.AccountKeeper
```

Comment:
I'm not sure we should be importing `cosmos-sdk/x/simulation` types in files outside `x/leverage/simulation` folder. We might need to find another way

Reply:
I could create an interface in the `x/leverage/types/expected_types.go` AccountKeeper

Reply:
Bez sent me this to look at https://github.com/cosmos/cosmos-sdk/blob/master/x/distribution/simulation/operations.go#L27 and indicated `we need access to the account and bank keeper but that's only in the app module type, not the keeper type`.

If mimicking that, we should not need to modify `x/leverage/keeper` and `x/leverage/types` folders at all I think.

In `operations_test.go` of the `cosmos-sdk/x/distribution` module, the following appears for example:
```
op := simulation.SimulateMsgSetWithdrawAddress(suite.app.AccountKeeper, suite.app.BankKeeper, suite.app.DistrKeeper)
```
so the account keeper is being retrieved from suite.app

Reply:
To check if I got it correctly it should add the AccountKeeper to the leverage `AppModule` structure instead of the `Keeper` 
Just like it happen here? https://github.com/cosmos/cosmos-sdk/blob/b5ca8e0b48279a74e126721929764d1ef09d3449/x/distribution/module.go#L204-L205
@toteki

Reply:
Yes, correct. There should be no reason to modify `x/leverage`'s `Keeper` type. Any external keeper dependencies needed for simulations should be added to the `AppModule` directly 👍